src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_NUMBER_OF_FIGURE = 4;
+    public static final int MAX_SIZE_OF_PARAMETER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Circle circle = new Circle("white", 10);
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(MAX_NUMBER_OF_FIGURE)) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE_OF_PARAMETER));
```

Comment:
Don't pass expressions in the constructor, instead create variables. See [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor).